### ISSUE TITLE
termux-fix-shebang : fails when incompatible cr/lf

### ISSUE THREAD

**Issue by rgh36167** [termux/termux-api#34]
happened to share a script by email (k9mail) and the attachment was mangled to contain "\r" at line ends.

termux-fix-shebang fixed the path but not the "\r", resulting in a rather "mysterious" error message.

dos2unix fixed it.

Btw.. isn't there some way to workaround to avoid needing to fix shebang paths at least for the primary shell?

**Comment by Neo-Oli** [comment]
The workaround is running termux-chroot, which places you in a virtual environment.

**Comment by rgh36167** [comment]
cool, didn't realize it would work without root.  Is there a trick to have the chroot called before starting scripts from termux-widget?

**Comment by Neo-Oli** [comment]
no, that is not possible. You can launch proot with a command argument, but you'd have to set up the environment yourself. Termux-chroot is just a wrapper script for proot. 
But proot is slower than a normal shell anyway and should be avoided if possible. Why do you actually need it. 
If it's a symlinking and compatibility issue, you could just have a wrapper script in .shortcuts.
For example, if you want to run a python script, you can just place a script in .shortcuts that contains
`python /path/to/script.py`

This completely circumvents the shebang.

**Comment by rgh36167** [comment]
thanks, that looks like the best possibility. Usually I am trying to have everything running identical on all devices but a small wrapper does not hurt.
